controller/Proctoring.php

```diff
@@ -0,0 +1,265 @@
+            $testCenter = $objectClass->createInstance('Dummy Test Center', 'temporarly generated test center', $fakeUri);
+        }
+        return $testCenter;
+    }
+
+    /**
+     * Get the requested test center resource
+     * Use this to identify which test center is currently being selected buy the proctor
+     *
+     * @return core_kernel_classes_Resource
+     * @throws \common_Exception
+     */
+    protected function getCurrentTestCenter()
+    {
+        if (is_null($this->currentTestCenter)) {
+            if($this->hasRequestParameter('testCenter')){
+
+                //@todo remove me
+                return $this->_getDummyTestCenter();
+
+                //get test center resource from its uri
+                $testCenterUri           = $this->getRequestParameter('testCenter');
+                $this->currentTestCenter = new core_kernel_classes_Resource($testCenterUri);
+            }else{
+                //@todo use a better exception
+                throw new \common_Exception('no current test center');
+            }
+            
+        }
+        return $this->currentTestCenter;
+    }
+
+    /**
+     * Get the requested delivery resource
+     * Use this to identify which delivery is currently being selected buy the proctor
+     * 
+     * @return core_kernel_classes_Resource
+     * @throws \common_Exception
+     */
+    protected function getCurrentDelivery()
+    {
+        if (is_null($this->currentDelivery)) {
+            if($this->hasRequestParameter('delivery')){
+
+                //@todo remove me
+                return $this->_getDummyDelivery();
+
+                //get test center resource from its uri
+                $deliveryUri           = $this->getRequestParameter('delivery');
+                $this->currentDelivery = new core_kernel_classes_Resource($deliveryUri);
+            }else{
+                //@todo use a better exception
+                throw new \common_Exception('no current delivery');
+            }
+        }
+        return $this->currentDelivery;
+    }
+
+    /**
+     * Main method to render a view for all proctoring related controller actions
+     * 
+     * @param string $cssClass
+     * @param array $data
+     * @param array $breadcrumbs
+     */
+    protected function composeView($cssClass, $data = array(), $breadcrumbs = array())
+    {
+        $data['breadcrumbs'] = $breadcrumbs;
+
+        if (!empty($_SERVER['HTTP_X_REQUESTED_WITH']) && strtolower($_SERVER['HTTP_X_REQUESTED_WITH']) == 'xmlhttprequest') {
+            $this->returnJson($data);
+        } else {
+            $this->defaultData();
+            $this->setData('userLabel', SessionManager::getSession()->getUserLabel());
+            $this->setData('clientConfigUrl', $this->getClientConfigUrl());
+            $this->setData('cls', $cssClass);
+            $this->setData('data', $data);
+            $this->setData('content-template', 'pages/index.tpl');
```

Comment:
Maybe can we put this as parameter, with default value to `pages/index.tpl` ?
Even if now there is no need for that, it can be useful for future needs